src/main/kotlin/org/rust/lang/refactoring/RustNameSuggestions.kt

```diff
@@ -0,0 +1,101 @@
+package org.rust.lang.refactoring
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.NameUtil
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.ide.navigation.goto.RustSymbolNavigationContributor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.childOfType
+import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.types.util.resolvedType
+import java.util.*
+
+/**
+ * This suggests names for an expression about to be bound to a local variable
```

Comment:
I'm not sure actually, I think there could be other cases, this is mainly done like this right now, due to the call site in the introduce local variable handler only having type information for psi element.